### PR TITLE
add ControlledBreaker model under Events sub-package

### PR DESCRIPTION
**Pull request by BiswarupM**

### CODE REVIEW

File: OpenIPSL/Electrical/Events/ControlledBreaker.mo [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  import Modelica.ComplexMath.j;
+  OpenIPSL.Interfaces.PwPin p annotation (
+    Placement(transformation(extent = {{-100, -10}, {-80, 10}}), iconTransformation(extent = {{-100, -10}, {-80, 10}})));
+  OpenIPSL.Interfaces.PwPin n annotation (
+    Placement(transformation(extent = {{80, -10}, {100, 10}}), iconTransformation(extent = {{80, -10}, {100, 10}})));
+  Complex vs(re = p.vr, im = p.vi);
+  Complex vr(re = n.vr, im = n.vi);
+  Complex is(re = p.ir, im = p.ii);
+  Complex ir(re = n.ir, im = n.ii);
+  Modelica.Blocks.Interfaces.BooleanInput TRIGGER annotation (
```
[the remark this answers is not in the document]
Comment (dietmarw) [reply OpenIPSL/Electrical/Events/ControlledBreaker.mo]: Also a description string is missing as to what TRIGGER is supposed to do.

**Comment by dietmarw** [comment]
Please document this model so that people know what to use it for.

**Comment by dietmarw** [comment]
So having looked at this model and at the existing `Breaker` then this controlled breaker should not be added but rather the existing `Breaker` should be extended with a conditional input. This way we don't get code duplication and can clean up the existing model at the same time.

In addition this model uses `n` and `p` for the pins while OpenIPSL normally uses `s` and `r` for the pins. Something that is confusing and should be adapted.

**Comment by dietmarw** [comment]
Just like to add that despite my comments looking all negative the addition is very valuable. I will create an improved version of breaker based on your model and you can have a look at it and say what you think.

**Comment by dietmarw** [comment]
OK I've added an improved version of `Breaker` which now also allows you to connect an external trigger to it. 
**Attention!** The model I've created is an **opening** breaker. So if triggered it will open the circuit and not closed as proposed in `ControlledBreaker` of this PR. Please check it out also there is a test model `OpenIPSL.Examples.Events.TestBreaker` to see it in action.

If you find this useful, then we can remove your model and merge this PR with my updated one.

**Comment by dietmarw** [comment]
@BiswarupM I'm answering here in order to keep it in place:

> 1. If you check the 'enableTrigger' box then you can trip it using some external command and If you don't then you have to set some time when you want to open the breaker.

That is correct.

> 2. If you check the 'rc_enable' box then you can re-close it using some external command and If you don't then you have to set some time when you want to re-close the breaker.

No, you can **EITHER** use the external trigger to open and close **OR** control the breaker with time parameters. Enabling `rc_enabled` will also allow you re-close the breaker at the time specified by `t_rc`.

> So for the OpenCPS models we have to check both the boxes to use the external command. 

As soon as you check `enableTrigger` you can control the breaker externally. It then no longer matters what  the other parameters and check boxes are set to since they are deactivated.

> We only need this model to replace previous Breaker and my controlled breaker model.

Yes.

**Comment by dietmarw** [comment]
Sorry I just noticed that my example model was not included. I've added that now:
 - `OpenIPSL.Examples.Events.TestBreaker`

Sorry for that. Here also a quick animation how the selection works: 
peek 2018-06-13 17-34

**Comment by BiswarupM** [comment]
nice example! Thanks!

**Comment by dietmarw** [comment]
So by this I understand you are fine with removing your model and extending the existing instead. Will do so in a minute.

**Pull request merged**